sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -251,6 +274,21 @@ def grow(self):
         while self.splittable_nodes:
             self.split_next()
 
+        self._apply_shrinkage()
+
+    def _apply_shrinkage(self):
+        """Multiply leaves values by shrinkage parameter.
+
+        This must be done at the very end of the growing process. If this were
+        done during the growing process e.g. in finalize_leaf(), then a leaf
+        would be shrunk but its sibling would potentially not be (if it's a
+        non-leaf), which would lead to a wrong computation of the 'middle'
+        value needed to enforce the monotonic constraints.
```

Comment:
That was a source of bug during development so I wrote this.

But it might actually just add confusion... LMK if I should remove.